### PR TITLE
Add some script size sanity checks

### PR DESCRIPTION
**Pull request by jonasschnelli**

### CODE REVIEW

**Comment by promag** [comment]
Weird branch name? 😛

File: src/script.c [hunk 1]
```diff
@@ -159,6 +159,11 @@ btc_bool btc_script_get_ops(const cstring* script_in, vector* ops_out)
             continue;
         }
 
+        // don't alloc a pushdata if there is no data
```
Comment (promag) [inline src/script.c]: Comment seems incomplete.

File: src/script.c [hunk 2]
```diff
@@ -159,6 +159,11 @@ btc_bool btc_script_get_ops(const cstring* script_in, vector* ops_out)
...
+        if (buf.len <= 0 || data_len > buf.len) {
```
Comment (promag) [inline src/script.c]: `len` is `size_t` so `buf.len == 0` is enough?

**Comment by jonasschnelli** [comment]
Thanks for the review @promag. Fixed your points.
Yes. Branch name is a bit recycled.

File: src/script.c [hunk 3]
```diff
@@ -159,6 +159,11 @@ btc_bool btc_script_get_ops(const cstring* script_in, vector* ops_out)
             continue;
         }
 
+        // don't alloc a push buffer if there is no more data available
+        if (buf.len == 0 || data_len > buf.len) {
```
Comment (promag) [inline src/script.c]: Looks like data_len is always `> 0` right? `if (data_len > buf.len)` should be enough then.

Comment (jonasschnelli) [reply src/script.c]: Could `deser_bytes` not de-serialize a 0 byte?

**Pull request merged**